File: mentalmadness.vb
Private Function mentalmadness(ByVal array As Integer(), ByVal size As Integer) As Integer
    Dim i As Integer
    Dim j As Integer
    Dim temp As Integer
    For i = 1 To size - 1
        temp = array(i)
        j = i - 1

        While j >= 0 AndAlso array(j) > temp
            array(j + 1) = array(j)
            j -= 1
        End While
        array(j + 1) = temp
    Next
    Return 0
End Function

File: mentalmadness.cpp
int mentalmadness(int *array, int size) {
    int i, j, temp;
    for (i = 1; i < size; i++) {
        temp = array[i];
        for (j = i - 1; j >= 0 && array[j] > temp; j--) {
            array[j + 1] = array[j];
        }
        array[j + 1] = temp;
    }
    return 0;
}
